### PR TITLE
Add tests using themis on github actions

### LINKED ISSUE #80: test: add CI and test
Add CI and test

### PR DESCRIPTION
**Pull request by Matts966**
# 💪 Summary

fix #80

- Added tests using themis on github actions.
- Currently, only the function for getting `gh` cli token is tested.

## 🏁 Goals

- Add test suite for CI
- Test token registration

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
```
Comment (skanehira) [inline .github/workflows/test.yml]: This CI name is Test

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# Controls when the action will run.
+on: push
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    strategy:
+      matrix:
+        vim_type: ['Vim'] # , 'Neovim']
```
Comment (skanehira) [inline .github/workflows/test.yml]: Is themis doesn't work on Neovim?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        version: ['head', 'stable']
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        include:
+          - vim_type: 'Vim'
+            version: 'stable'
+            vim_version: 'v8.2.0000'
+    runs-on: '${{ matrix.os }}'
+    steps:
+      - uses: 'actions/checkout@v2'
+      - uses: 'actions/checkout@v2'
```
Comment (skanehira) [inline .github/workflows/test.yml]: checkout is duplicated.

File: autoload/gh/gh.vim [hunk 4]
```diff
@@ -229,3 +229,18 @@ function! gh#gh#decide_open() abort
   endif
   return ''
 endfunction
+
+function! gh#gh#get_token() abort
```
Comment (skanehira) [inline autoload/gh/gh.vim]: This is already merged in other PR.
Please rebase master branch.

**Review by skanehira — CHANGES_REQUESTED** [review]
Thanks!
I have some comment.

File: .github/workflows/test.yml (hunk 2 above)
Comment (Matts966) [reply .github/workflows/test.yml]: I tested yesterday, but it fails when building (not related to test).
I am now trying again!

File: .github/workflows/test.yml (hunk 3 above)
Comment (Matts966) [reply .github/workflows/test.yml]: This seems required. The first one is for this repo and second one is for `themis`.

File: .github/workflows/test.yml (hunk 2 above)
Comment (Matts966) [reply .github/workflows/test.yml]: Tested but failed on ubuntu-latest with Neovim stable.
https://github.com/Matts966/gh.vim/actions/runs/523346511

It seems odd but there are no clear messages. I amended to remove `Neovim`.

Please notify me if it is desirable for further investigation.

Comment (skanehira) [reply .github/workflows/test.yml]: How about this action?
It seems like just download executable binary
https://github.com/rhysd/action-setup-vim

**Review by skanehira — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (Matts966) [reply .github/workflows/test.yml]: Thank you! I will try it by today 👍

Comment (Matts966) [reply .github/workflows/test.yml]: It worked fine! Tested also on Neovim. Thank you for your advice.
https://github.com/Matts966/gh.vim/actions/runs/525275004

Comment (skanehira) [reply .github/workflows/test.yml]: Sounds good.

File: test/gh/gh.vim [hunk 5]
```diff
@@ -0,0 +1,7 @@
+let s:suite = themis#suite('gh')
+let s:assert = themis#helper('assert')
+
+function! s:suite.test_get_gh_token()
+  let gh_token = gh#gh#get_token()
+  call s:assert.false(empty(gh_token))
```
Comment (skanehira) [inline test/gh/gh.vim]: We have to add test case which gh doesn't used by user.

**Review by skanehira — CHANGES_REQUESTED** [review]

**Comment by Matts966** [comment]
@skanehira 
Added cases without `gh` and with `g:gh_token` :)

**Review by skanehira — APPROVED** [review]
LGTM
Thanks :)

**Pull request merged**